Updated setup file

DrGuy opened the pull request:
Replaced PyGDAL with GDAL.
Added functions to update ENVI headers.
Fixed bug relating to dictionary handling for Landsat 4-7 data.
Minor changes relating to updated 3.3 Matlab code.

Commit:
Updated setup file

Replaced PyGDAL with GDAL

robintw commented:
Thanks for the changes - I'll definitely merge them shortly.

I'm intrigued by why the changes for the stack dictionary were needed - can you elaborate? The new code seems to have exactly the same functionality as the old.

DrGuy commented:
There was an issue with the way the stack dictionary was merged with the
ESUN list. Also, numexpr.evaluate was having problems with the im_B?
variables, as they weren't in the stack dictionary and it wasn't accessing
global data.

On Wed, Dec 2, 2015 at 12:49 PM, Robin Wilson notifications@github.com
wrote:

> Thanks for the changes - I'll definitely merge them shortly.
> 
> I'm intrigued by why the changes for the stack dictionary were needed -
> can you elaborate? The new code seems to have exactly the same
> functionality as the old.
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/robintw/PyFMask/pull/2#issuecomment-161281769.

robintw commented:
Hmm, interesting: I'll do some tests myself and just check it all works, but assuming it does then I'll be happy to merge it.

Out of interest, what version(s) of Python have you tested it on? I wonder if `locals()` behaves slightly differently on different versions.

DrGuy commented:
Python 3.4.3 :: Anaconda 2.1.0 (64-bit)

What version are you using?

On Wed, Dec 2, 2015 at 1:01 PM, Robin Wilson notifications@github.com
wrote:

> Hmm, interesting: I'll do some tests myself and just check it all works,
> but assuming it does then I'll be happy to merge it.
> 
> Out of interest, what version(s) of Python have you tested it on? I wonder
> if locals() behaves slightly differently on different versions.
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/robintw/PyFMask/pull/2#issuecomment-161284737.

DrGuy commented:
Also, it's running on an Ubuntu 64-bit VM.

On Wed, Dec 2, 2015 at 1:03 PM, Guy Serbin guy.serbin@gmail.com wrote:

> Python 3.4.3 :: Anaconda 2.1.0 (64-bit)
> 
> What version are you using?
> 
> On Wed, Dec 2, 2015 at 1:01 PM, Robin Wilson notifications@github.com
> wrote:
> 
> > Hmm, interesting: I'll do some tests myself and just check it all works,
> > but assuming it does then I'll be happy to merge it.
> > 
> > Out of interest, what version(s) of Python have you tested it on? I
> > wonder if locals() behaves slightly differently on different versions.
> > 
> > —
> > Reply to this email directly or view it on GitHub
> > https://github.com/robintw/PyFMask/pull/2#issuecomment-161284737.

DrGuy commented:
Another heads up- rather than use the imfill() function in fmask.py, I
modded the code to use scipy.ndimage.morphology.binary_fill_holes()
instead. I'll commit the changes after I have a chance to further review
the code and make sure it's properly producing the files.

On Wed, Dec 2, 2015 at 1:04 PM, Guy Serbin guy.serbin@gmail.com wrote:

> Also, it's running on an Ubuntu 64-bit VM.
> 
> On Wed, Dec 2, 2015 at 1:03 PM, Guy Serbin guy.serbin@gmail.com wrote:
> 
> > Python 3.4.3 :: Anaconda 2.1.0 (64-bit)
> > 
> > What version are you using?
> > 
> > On Wed, Dec 2, 2015 at 1:01 PM, Robin Wilson notifications@github.com
> > wrote:
> > 
> > > Hmm, interesting: I'll do some tests myself and just check it all works,
> > > but assuming it does then I'll be happy to merge it.
> > > 
> > > Out of interest, what version(s) of Python have you tested it on? I
> > > wonder if locals() behaves slightly differently on different versions.
> > > 
> > > —
> > > Reply to this email directly or view it on GitHub
> > > https://github.com/robintw/PyFMask/pull/2#issuecomment-161284737.

DrGuy commented:
One other thing- my version corrupts the MTL file- DO NOT USE! I fixed the
bug, but have some more debugging to do.

On Wed, Dec 2, 2015 at 4:22 PM, Guy Serbin guy.serbin@gmail.com wrote:

> Another heads up- rather than use the imfill() function in fmask.py, I
> modded the code to use scipy.ndimage.morphology.binary_fill_holes()
> instead. I'll commit the changes after I have a chance to further review
> the code and make sure it's properly producing the files.
> 
> On Wed, Dec 2, 2015 at 1:04 PM, Guy Serbin guy.serbin@gmail.com wrote:
> 
> > Also, it's running on an Ubuntu 64-bit VM.
> > 
> > On Wed, Dec 2, 2015 at 1:03 PM, Guy Serbin guy.serbin@gmail.com wrote:
> > 
> > > Python 3.4.3 :: Anaconda 2.1.0 (64-bit)
> > > 
> > > What version are you using?
> > > 
> > > On Wed, Dec 2, 2015 at 1:01 PM, Robin Wilson notifications@github.com
> > > wrote:
> > > 
> > > > Hmm, interesting: I'll do some tests myself and just check it all
> > > > works, but assuming it does then I'll be happy to merge it.
> > > > 
> > > > Out of interest, what version(s) of Python have you tested it on? I
> > > > wonder if locals() behaves slightly differently on different versions.
> > > > 
> > > > —
> > > > Reply to this email directly or view it on GitHub
> > > > https://github.com/robintw/PyFMask/pull/2#issuecomment-161284737.

Commit:
Bug fixes

1. Removed references to imfill() routine, replaced with
scipy.ndimage.morphology.binary_fill_holes()
2. Fixed bug that corrupted MTL file.
3. Fixed bugs in ENVI header revision routine.

Commit:
Bug fixes

Commit:
Further debugging

Commit:
Further bug fixes

This still isn't operational

Commit:
Further revisions, still buggy

1. Unused function deleted.
2. Statistics functions from scipy.stats and numpy modified to behave
like in Matlab.
3. Bug still present whereby cloud heights are grossly overestimated.

DrGuy commented:
I fixed the issue mentioned earlier, and made some minor changes. I also deleted unused functions. However, this is still grossly overestimating cloud heights, resulting in cloud shadows being identified far from their actual locations.

DrGuy commented:
I've fixed the MTL file corruption issue and made a few other corrections:
1. Deleted unused functions to minimize confusion.
2. Added functions to fix ENVI header file for main Fmask output.
3. Modified statistics function to behave like their Matlab equivalents.

One huge bug- the cloud heights are grossly overestimated, so cloud shadows are misplaced, and I am unsure how to fix this.

robintw commented:
Thanks for your work on this.

Can I just confirm: is the bug with the cloud heights present in the original code, or just in the altered code in this Pull Request?

DrGuy commented:
You're welcome. It's also in the original code. I am unsure as to whether
it's propagated from 3.2 to 3.3- I did my best to ensure that the code is
compliant with the latest version. Also, I have only tested it on Windows 7
using a 64-bit version of Miniconda Python 3.4.

On Wed, Jan 13, 2016 at 7:03 PM, Robin Wilson notifications@github.com
wrote:

> Thanks for your work on this.
> 
> Can I just confirm: is the bug with the cloud heights present in the
> original code, or just in the altered code in this Pull Request?
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/robintw/PyFMask/pull/2#issuecomment-171399278.
